BridgeCareApp/BridgeCare/DataAccessLayer/InvestmentStrategies.cs

```diff
@@ -18,25 +21,101 @@ public InvestmentStrategies(BridgeCareContext context)
 
         public IQueryable<InvestmentStrategyModel> GetInvestmentStrategies(NetworkModel network, BridgeCareContext db)
         {
-            var investmentStrategies = Enumerable.Empty<InvestmentStrategyModel>();
             try
             {
-                investmentStrategies = db.INVESTMENTs
-                    .Select(p => new InvestmentStrategyModel
+                var simulation = db.SIMULATIONS
+                    .Include(d => d.INVESTMENTS)
+                    .Include(d => d.YEARLYINVESTMENTs)
+                    .Select(p => new InvestmentStrategyModel()
                     {
-                        Id = p.SIMULATIONID,
-                        BudgetOrder = p.BUDGETORDER,
-                        FirstYear = p.FIRSTYEAR ?? 0,
-                        NumberYears = p.NUMBERYEARS ?? 0,
-                        InflationRate = p.INFLATIONRATE ?? 0,
-                        DiscountRate = p.DISCOUNTRATE ?? 0
-                    });
+                        Name = p.SIMULATION1,
+                        Description = p.COMMENTS,
+                        SimulationId = p.SIMULATIONID,
+                        NetworkId = p.NETWORKID ?? 0,
+                        FirstYear = p.INVESTMENTS.FIRSTYEAR ?? 0,
+                        NumberYears = p.INVESTMENTS.NUMBERYEARS ?? 0,
+                        InflationRate = p.INVESTMENTS.INFLATIONRATE ?? 0,
+                        DiscountRate = p.INVESTMENTS.DISCOUNTRATE ?? 0,
+                        BudgetOrder = p.INVESTMENTS.BUDGETORDER,
+                        YearlyBudget = p.YEARLYINVESTMENTs.Select(m => new InvestmentStrategyYearlyBudgetModel
+                        {
+                            Year = m.YEAR_,
+                            Budget = p.YEARLYINVESTMENTs
+                            .Where(n => n.YEAR_ == m.YEAR_)
+                            .Select(f => new InvestmentStrategyBudgetModel()
+                            {
+                                Amount = f.AMOUNT,
+                                Name = f.BUDGETNAME
+                            }).ToList()
```

Comment:
Is it necessary to do this ToList conversion three times?

Reply:
I think yes. Cannot devise improvement, this is fairly condensed code,  the 3 ToList()'s are a function of the destination class. Closest similar queries I can find in our code base span multiple queries and more lines of code for the same functionality so the folding into the structure happens in a more distributed way. My understanding is we want to condense code when possible.